Create ACTIONS - UNREIMBURSED UNINSURED RETURNED DOCS.vbs

gretchent74 opened the pull request:
#441 resolves issue #441
separated unreimbursed uninsured documents sent to participant and documents received from participants.  this script deals with process when a party sends in a request to collect unreimbursed/uninsured expenses.  this script will create a caad note of documents returned, send 2 dord documents to the non requesting party, and create jude, obbd, ncod or cpod.

theVKC commented on Script Files/ACTIONS/ACTIONS - UNREIMBURSED UNINSURED RETURNED DOCS.vbs:
```diff
@@ -0,0 +1,751 @@
+
+
+'this is a function document
+DIM beta_agency, FuncLib_URL, run_locally, use_master_branch, req, fso
```
Please clear this out. No need to DIM.

theVKC commented on Script Files/ACTIONS/ACTIONS - UNREIMBURSED UNINSURED RETURNED DOCS.vbs:
```diff
@@ -0,0 +1,751 @@
+'this is where the copy and paste from functions library ended
+
+'first dialog 
+DIM UnUn1_Dialog, PRISM_case_number, CP_checkbox, NCP_checkbox, err_msg, ButtonPressed
```
Please clear this out. No need to DIM.

theVKC commented on Script Files/ACTIONS/ACTIONS - UNREIMBURSED UNINSURED RETURNED DOCS.vbs:
```diff
@@ -0,0 +1,751 @@
+CALL navigate_to_PRISM_screen ("CAPS")
+
+'this auto fills prism case number in dialog
+EMReadScreen PRISM_case_number, 13, 4, 8 
```
Can you switch to using the `PRISM_case_number_finder` function?:

`call PRISM_case_number_finder(PRISM_case_number)`

That function should find the PRISM case number even if it ever moves on the screen. It also standardizes the process.

theVKC commented on Script Files/ACTIONS/ACTIONS - UNREIMBURSED UNINSURED RETURNED DOCS.vbs:
```diff
@@ -0,0 +1,751 @@
+
+
+'dialog if cp is requesting collection of un/un
+DIM Cp_requested_Dialog, CPCAAD_checkbox, worker_signature, jude_checkbox, amount, ncp_documents_checkbox
```
No need for the dialog DIM or worker_signature DIMing. Clearly you learned to use Option Explicit well, and that's a great thing! Unfortunately it doesn't work so slick for scripts once they're on GitHub (so you wouldn't know it from your own testing, unfortunately).

theVKC commented on Script Files/ACTIONS/ACTIONS - UNREIMBURSED UNINSURED RETURNED DOCS.vbs:
```diff
@@ -0,0 +1,751 @@
+EndDialog
+
+'dialog box for date on aff of service
+DIM date, DATE_SERVED_dialog, date_served, confidential_checkbox
```
I think you can get rid of all of this.

theVKC commented on Script Files/ACTIONS/ACTIONS - UNREIMBURSED UNINSURED RETURNED DOCS.vbs:
```diff
@@ -0,0 +1,751 @@
+	transmit
+
+'''need to select legal heading
+DIM LH_dialog
```
You can get rid of this.

theVKC commented on Script Files/ACTIONS/ACTIONS - UNREIMBURSED UNINSURED RETURNED DOCS.vbs:
```diff
@@ -0,0 +1,751 @@
+' This is a custom function to change the format of a participant name.  The parameter is a string with the 
+' client's name formatted like "Levesseur, Wendy K", and will change it to "Wendy K LeVesseur".  
+
+FUNCTION change_client_name_to_FML(client_name)
```
I can't imagine you need these custom functions anymore... Did you try without?

theVKC commented on Script Files/ACTIONS/ACTIONS - UNREIMBURSED UNINSURED RETURNED DOCS.vbs:
```diff
@@ -0,0 +1,751 @@
+  Text 120, 140, 50, 10, "xx/xx/xxxx"
+EndDialog
+
+DIM PRISM_INFO_Dialog, Co_Seq, From_date, To_date, cp_name, eff_date, beg_date
```
I think all this can go.

theVKC commented on Script Files/ACTIONS/ACTIONS - UNREIMBURSED UNINSURED RETURNED DOCS.vbs:
```diff
@@ -0,0 +1,751 @@
+	EMWriteScreen "C", 3, 29
+	transmit
+	EMWriteScreen "A", 3, 29
+	EMWriteScreen (Co_Seq), 4, 34
```
The parenthesis in these areas don't need to be here. EMWriteScreen doesn't require it in a parameter.

theVKC commented on Script Files/ACTIONS/ACTIONS - UNREIMBURSED UNINSURED RETURNED DOCS.vbs:
```diff
@@ -0,0 +1,751 @@
+	EMWriteScreen "A", 3, 29
+	EMWriteScreen "JME", 4, 34
+	EMWriteScreen "  ", 4, 053
+	EMWriteScreen (eff_date), 9, 59 
```
Same as above.

theVKC commented on Script Files/ACTIONS/ACTIONS - UNREIMBURSED UNINSURED RETURNED DOCS.vbs:
```diff
@@ -0,0 +1,751 @@
+	transmit
+	'transmit
+	
+	DIM ncod_success
```
You can get rid of this one also.

theVKC commented on Script Files/ACTIONS/ACTIONS - UNREIMBURSED UNINSURED RETURNED DOCS.vbs:
```diff
@@ -0,0 +1,751 @@
+  Text 35, 155, 65, 10, "Affidavit of Service"
+EndDialog
+
+DIM Ncp_requested_Dialog, NCPCAAD_checkbox, cpod_checkbox, cp_documents_checkbox
```
This too can go.

theVKC commented on Script Files/ACTIONS/ACTIONS - UNREIMBURSED UNINSURED RETURNED DOCS.vbs:
```diff
@@ -0,0 +1,751 @@
+	EMWriteScreen "S", 11, 5
+	transmit 
+	
+	EMWriteScreen(amount), 16, 15
```
No need for the parenthesis.

theVKC commented on Script Files/ACTIONS/ACTIONS - UNREIMBURSED UNINSURED RETURNED DOCS.vbs:
```diff
@@ -0,0 +1,751 @@
+	transmit
+	EMWriteScreen "Medical/Dental Expenses", 16, 15
+	transmit
+	EMWriteScreen (date_served), 16, 15
```
No need for the parenthesis.

theVKC commented on Script Files/ACTIONS/ACTIONS - UNREIMBURSED UNINSURED RETURNED DOCS.vbs:
```diff
@@ -0,0 +1,751 @@
+			Dialog LH_dialog  'name of dialog
+			IF buttonpressed = 0 then stopscript		'Cancel
+			
+END IF
```
These `END IF` statements look like they're a bit reversed. The indenting is a bit backwards.

theVKC commented on Script Files/ACTIONS/ACTIONS - UNREIMBURSED UNINSURED RETURNED DOCS.vbs:
```diff
@@ -0,0 +1,751 @@
+
+IF cpod_checkbox = 1 THEN
+
+CALL navigate_to_PRISM_screen ("SUOD")
```
All of this should be indented...

theVKC commented on Script Files/ACTIONS/ACTIONS - UNREIMBURSED UNINSURED RETURNED DOCS.vbs:
```diff
@@ -0,0 +1,751 @@
+			MsgBox "***NOTICE!!!***" & vbNewline & err_msg & vbNewline & vbNewline & "Please resolve for the script to continue."
+		END IF
+
+Loop until err_msg = ""
```
From above, you should indent all of this down one.

theVKC commented on Script Files/ACTIONS/ACTIONS - UNREIMBURSED UNINSURED RETURNED DOCS.vbs:
```diff
@@ -0,0 +1,751 @@
+			Msgbox "OBBD information was not added correctly, please reneter information.  Script Ended."
+			StopScript
+		END IF
+CALL navigate_to_PRISM_screen ("CPOL")
```
This should also be indented.

theVKC commented on Script Files/ACTIONS/ACTIONS - UNREIMBURSED UNINSURED RETURNED DOCS.vbs:
```diff
@@ -0,0 +1,751 @@
+	'DIM obbd_success
+	EMReadScreen obbd_success, 13 , 24, 66
+		IF obbd_success <> "modified succ" THEN 
+			Msgbox "OBBD information was not added correctly, please reneter information.  Script Ended."
```
Please replace these error messages with `script_end_procedure("")`, putting the MsgBox message inside the quotes. This allows agencies-running-usage-statistics to see the errors (helpful for us when we are debugging).

theVKC commented on Script Files/ACTIONS/ACTIONS - UNREIMBURSED UNINSURED RETURNED DOCS.vbs:
```diff
@@ -0,0 +1,751 @@
+	DIM cpod_success
+	EMReadScreen cpod_success, 18 , 24, 33
+		IF cpod_success <> "added successfully" THEN 
+			Msgbox "CPOD information was not added correctly, please reneter information.  Script Ended."
```
Please replace these error messages with `script_end_procedure("")`, putting the MsgBox message inside the quotes. This allows agencies-running-usage-statistics to see the errors (helpful for us when we are debugging).

gretchent74 commented:
fixed DIM issues and corrected indenting.
separated unreimbursed uninsured documents sent to participant and documents received from participants. this script deals with process when a party sends in a request to collect unreimbursed/uninsured expenses. this script will create a caad note of documents returned, send 2 dord documents to the non requesting party, and create jude, obbd, ncod or cpod.
